packages/core/src/Project.ts

```diff
@@ -170,6 +182,7 @@ export class Project {
   private previousBufferContext: CanvasRenderingContext2D | null = null;
   private exportCounter = 0;
   private size = Vector2.zero;
+  private variables: {[key: string]: any} = {};
```

Comment:
I would suggest that we use a type like `unknown` here instead of `any` -- `any` turns off the type system, and since we return this to user-code that means we can pollute the typing in that code, too. The way using `unknown` would require users to ensure that the project variables are the types they are expecting (or they can cast them if they so choose, but then they're making the choice).

Reply:
Btw, should be `Record<string, unknown>` to stay consistent with the rest of the codebase.